src/OrchardCore.Modules/OrchardCore.ContentLocalization/Startup.cs

```diff
@@ -56,7 +57,6 @@ public override void ConfigureServices(IServiceCollection services)
                 options.RequestCultureProviders.Insert(0, new ContentRequestCultureProvider());
             });
 
-            services.AddScoped<IContentPartHandler, LocalizationPartHandler>();
             services.AddSingleton<ILocalizationEntries, LocalizationEntries>();
```

Comment:
Unless I am missing something, you need to move `services.AddSingleton<ILocalizationEntries, LocalizationEntries>();` as well or the `LocalizationPartHandler` will throw an exception if the Content Culture Picker feature is turned off?

Or in the handler you resolve `ILocalizationEntries` from the service provider instead of constructor and null check it in the `PublishedAsync` methods etc